### PR TITLE
Reduce masternode collateral amount

### PR DESCRIPTION
**Pull request by bvbfan**
It's used in POS as well.
Fixes https://github.com/cakedefi/defichain-private/issues/339

### CODE REVIEW

**Comment by bvbfan** [comment]
Rebase to 1.5.x should be OK, now.

**Comment by monstrobishi** [comment]
Can I confirm that existing masternodes that staked 1 million will be able to continue staking 1 million until they unstake and stake again? We can check creation height against the hardfork height

**Comment by bvbfan** [comment]
It makes sense.

**Comment by bvbfan** [comment]
Force push changes to keep node created before Dakota height to keep their stakes.

File: src/test/miner_tests.cpp [hunk 1]
```diff
@@ -250,13 +251,14 @@ BOOST_AUTO_TEST_CASE(CreateNewBlock_validity)
 //                BOOST_CHECK(nodePtr && nodePtr->IsActive(tip->height));
 //
 //                mintedBlocks = nodePtr->mintedBlocks;
+//                creationHeight = int64_t(nodePtr->creationHeight);
 //            }
```
Comment (ShengguangXiao) [inline src/test/miner_tests.cpp:255]: Can remove all these commented code

Comment (bvbfan) [reply src/test/miner_tests.cpp:255]: Probably we can enabled it in the future?

**Pull request merged**